controller/ProviderAdmin.php

```diff
@@ -41,4 +44,32 @@ public function getClassService()
     {
         return $this->getServiceLocator()->get(RdfLtiProviderRepository::class);
     }
+
+    protected function getExtraValidationRules(): array
+    {
+        return $this->getValidationFactory()->createFormValidators($this->getLtiVersion());
+    }
+
+    private function getValidationFactory(): ValidatorsFactory
```

Comment:
Are you creating multiple validators here?

Reply:
I agree the name of the factory should be `ValidatorFactory` (singular) and the factory method could be `createMultiple...`.